File: mobile application development/front/react-native/src/api/authApi.ts
import { apiRequest } from './client';
import { API_PREFIX } from '../config/api';
import type {
  AuthResponse,
  AvailabilityResponse,
  LoginRequest,
  RegisterRequest,
} from './types';

export function registerApi(body: RegisterRequest) {
  return apiRequest<AuthResponse>(`${API_PREFIX}/auth/register`, {
    method: 'POST',
    body,
    auth: false,
  });
}

export function loginApi(body: LoginRequest) {
  return apiRequest<AuthResponse>(`${API_PREFIX}/auth/login`, {
    method: 'POST',
    body,
    auth: false,
  });
}

export function logoutApi(jwt: string) {
  return apiRequest<void>(`${API_PREFIX}/auth/logout`, { method: 'POST', jwt });
}

export function checkUsernameInUseApi(username: string) {
  return apiRequest<AvailabilityResponse>(`${API_PREFIX}/auth/checkUsernameInUse`, {
    method: 'POST',
    body: { username },
    auth: false,
  });
}

export function checkEmailInUseApi(email: string) {
  return apiRequest<AvailabilityResponse>(`${API_PREFIX}/auth/checkEmailInUse`, {
    method: 'POST',
    body: { email },
    auth: false,
  });
}

File: mobile application development/front/kotlin/app/src/main/java/com/itmo/mybroker/api/AuthApi.kt
package com.itmo.mybroker.api

object AuthApi {
    private const val PREFIX = "/api/broker-app/v1"

    fun register(body: RegisterRequest): AuthResponse =
        ApiClient.request("$PREFIX/auth/register", method = "POST", body = body, auth = false)

    fun login(body: LoginRequest): AuthResponse =
        ApiClient.request("$PREFIX/auth/login", method = "POST", body = body, auth = false)

    fun logout() {
        ApiClient.postEmpty("$PREFIX/auth/logout")
    }

    fun checkUsernameInUse(username: String): AvailabilityResponse =
        ApiClient.request(
            "$PREFIX/auth/checkUsernameInUse",
            method = "POST",
            body = mapOf("username" to username),
            auth = false,
        )

    fun checkEmailInUse(email: String): AvailabilityResponse =
        ApiClient.request(
            "$PREFIX/auth/checkEmailInUse",
            method = "POST",
            body = mapOf("email" to email),
            auth = false,
        )
}
